File: Node/Prisma_ORM/src/bll/services/orderDetailService.js
const OrderDetailRepository = require('../../dal/repositories/orderDetailRepository');

const OrderDetailService = {
  getAll: async () => {
    return await OrderDetailRepository.getAll();
  },

  getById: async (orderID, productID) => {
    const entity = await OrderDetailRepository.getById(orderID, productID);
    if (!entity) throw new Error('entity not found');
    return entity;
  },

  create: async (entity) => {
    return await OrderDetailRepository.create(entity);
  },

  delete: async (orderID, productID) => {
    return await OrderDetailRepository.delete(orderID, productID);
  }
};

module.exports = OrderDetailService; 

File: Go/GORM_ORM/bll/services/orderDetailService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type OrderDetailService interface {
	GetAll() ([]entities.OrderDetail, error)
	GetByID(orderId int, productId int) (*entities.OrderDetail, error)
	Create(orderDetail *entities.OrderDetail) error
	Delete(orderId int, productId int) error
}

type orderDetailService struct {
	repo repositories.OrderDetailRepository
}

func NewOrderDetailService(repo repositories.OrderDetailRepository) OrderDetailService {
	return &orderDetailService{repo: repo}
}

func (s *orderDetailService) GetAll() ([]entities.OrderDetail, error) {
	return s.repo.GetAll()
}

func (s *orderDetailService) GetByID(orderId int, productId int) (*entities.OrderDetail, error) {
	return s.repo.GetByID(orderId, productId)
}

func (s *orderDetailService) Create(orderDetail *entities.OrderDetail) error {
	return s.repo.Create(orderDetail)
}

func (s *orderDetailService) Delete(orderId int, productId int) error {
	return s.repo.Delete(orderId, productId)
}
